test/registry/NPMRegistryClient.test.ts

```diff
@@ -228,110 +145,298 @@ describe('NPMRegistryClient', () => {
+            } else if (
+              uri ===
+              'https://packages.fhir.org/hl7.terminology.r4/-/hl7.terminology.r4-no-versions.tgz'
+            ) {
+              return {
+                status: 200,
+                data: Readable.from(['no-versions-test-data'])
+              };
+            } else if (
+              uri ===
+              'https://packages.fhir.org/hl7.terminology.r4.no.tarball/-/hl7.terminology.r4.no.tarball-no-tarball-version.tgz'
+            ) {
+              return {
+                status: 200,
+                data: Readable.from(['no-tarball-test-data'])
+              };
+            } else if (
+              uri ===
+              'https://packages.fhir.org/hl7.terminology.no-dist/-/hl7.terminology.no-dist-no-dist-version.tgz'
+            ) {
+              return {
+                status: 200,
+                data: Readable.from(['no-dist-test-data'])
+              };
+            } else {
+              throw new Error('Not found');
+            }
+          });
+        });
 
-      it('should get the package using a created tgz file path when has manifest but has no versions', async () => {
-        const latest = await client.download('hl7.terminology.r4', 'no-versions');
-        expect(loggerSpy.getLastMessage('info')).toBe(
-          'Attempting to download hl7.terminology.r4#no-versions from https://packages.fhir.org/hl7.terminology.r4/-/hl7.terminology.r4-no-versions.tgz'
-        );
-        expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
-        expect(latest).toBeDefined();
-        expect(latest).toBeInstanceOf(Readable);
-        expect(latest.read()).toBe('no-versions-test-data');
-      });
+        afterAll(() => {
+          axiosSpy.mockRestore();
+        });
 
-      it('should get the package using a created tgz file path when has manifest but not correct version', async () => {
-        const latest = await client.download('hl7.terminology.r4', '1.1.5');
-        expect(loggerSpy.getLastMessage('info')).toBe(
-          'Attempting to download hl7.terminology.r4#1.1.5 from https://packages.fhir.org/hl7.terminology.r4/-/hl7.terminology.r4-1.1.5.tgz'
-        );
-        expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
-        expect(latest).toBeDefined();
-        expect(latest).toBeInstanceOf(Readable);
-        expect(latest.read()).toBe('1.1.5-test-data');
-      });
+        it('should get the package using tarball when has manifest tarball url', async () => {
+          const latest = await client.download('hl7.terminology.r4', '1.2.3-test');
+          expect(loggerSpy.getLastMessage('info')).toBe(
+            'Attempting to download hl7.terminology.r4#1.2.3-test from https://packages.simplifier.net/hl7.terminology.r4/1.2.3-test'
+          );
+          expect(loggerSpy.getAllMessages('error')).toHaveLength(0);
+          expect(latest).toBeInstanceOf(Readable);
+          expect(latest.read()).toBe('1.2.3-test-data');
+        });
```

Comment:
Is this testing the same thing as the first `#DownloadURL` test on line 399? I think it is.

I'm assuming this `#CreateURL` subsuite is for the case when there isn't a tarball URL in the manifest, so FPL creates one based on the NPM convention. This test doesn't fit into that category anyway, so it probably doesn't belong here. If it's basically the same as the one in `#DownloadURL`, you can probably just delete it.